src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs

```diff
@@ -1068,17 +1068,33 @@ public void WriteBufferLines(BufferChar[] buffer, ref int top, bool ensureBottom
 
             Console.SetCursorPosition(0, (top>=0) ? top : 0);
 
+            var lastForegroundColor = (ConsoleColor)(-1);
+            var lastBackgroundColor = (ConsoleColor)(-1);
             for (int i = 0; i < buffer.Length; ++i)
             {
                 // TODO: use escape sequences for better perf
-                Console.ForegroundColor =  buffer[i].ForegroundColor;
-                Console.BackgroundColor =  buffer[i].BackgroundColor;
+                var nextForegroundColor = buffer[i].ForegroundColor;
+                var nextBackgroundColor = buffer[i].BackgroundColor;
+                if (lastForegroundColor != nextForegroundColor)
+                {
+                    Console.ForegroundColor = lastForegroundColor = nextForegroundColor;
+                }
+                if (lastBackgroundColor != nextBackgroundColor)
+                {
+                    Console.BackgroundColor = lastBackgroundColor = nextBackgroundColor;
+                }
 
                 Console.Write(buffer[i].UnicodeChar);
             }
 
-            Console.BackgroundColor = backgroundColor;
-            Console.ForegroundColor = foregroundColor;
+            if (backgroundColor != lastBackgroundColor)
```

Comment:
Pretty much everywhere else you have foreground first, then background.  Another nitpick, but I would reorder these two `if` statements